### PR TITLE
refactor: core/sdk/clients split

### PR DESCRIPTION
**Pull request by geolffreym**
refactor: split core and sdk layers
test: chain types and implementations
docs: add we3 specifications

### CODE REVIEW

File: src/core/tests/web3/test_factory.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    """Should return valid w3 wrapper with valid provider"""
+
+    rinkeby = chain(4)
+    kovan = chain(42)
+
+    assert isinstance(rinkeby, Rinkeby)
+    assert isinstance(kovan, Kovan)
+
+
+def test_w3_invalid_provider():
```
Comment (geolffreym) [inline src/core/tests/web3/test_factory.py:56]: Add test for positive test case

File: src/core/storage/ipfs.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import json
+
+from ..exception import IPFSFailedExecution
+from ..util import resolve_root_for
+from ..constants import IPFS_CONTAINER
+
+ipfs = "ipfs"
+
+
+def get_container():
```
Comment (geolffreym) [inline src/core/storage/ipfs.py:13]: Add docs

File: src/core/storage/ipfs.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    Retrieve dag information from cid
+    Proxy dag get command to node
+    http://docs.ipfs.io.ipns.localhost:8080/reference/cli/#ipfs-dag-get
+    :param cid:
+    """
+    return exec_command("/dag/get", cid)
+
+
+def get_id():
+    output = exec_command("id")
```
Comment (geolffreym) [inline src/core/storage/ipfs.py:87]: Add docs

File: src/core/storage/ipfs.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
+import os
+import errno
+import docker
+import json
+
+from ..exception import IPFSFailedExecution
+from ..util import resolve_root_for
+from ..constants import IPFS_CONTAINER
+
+ipfs = "ipfs"
```
Comment (aphelionz) [inline src/core/storage/ipfs.py:10]: maybe move this to constants?

**Review by aphelionz — APPROVED** [review]

**Pull request merged**